[오상구] Refactor: 수정요구사항 반영

SangkuOh opened the pull request:
wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
필수 구현사항 구현 완료
		 
## 수정 사항들 자세한 내용
- 몬스터 API 호출 후 state 저장 완료
- CardList 컴포넌트 monsters props로 monster 배열 전달 완료
- Card 컴포넌트에 이미지주소, name, email 경로 수정 완료
- CardList에서 map 함수로 Card 컴포넌트 리턴완료
- Monster 컴포넌트 hanldeChange 메소드 자식에게 props 전달 후 setState
기능 구현 완료
- 변수에 필터링 로직 넣어서 값 비교후 조건에 맞는 카드만 보이게 구현완료

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- 함수 로직이 너무 긴 것 같습니다. 좀 더 효율적인 방법이 없을지 궁금합니다.(예시) 

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

jamessoun93 commented on src/2-monsters/index.js:
```diff
@@ -23,21 +23,33 @@ import "./index.css";
+  componentDidMount() {
+    fetch('https://jsonplaceholder.typicode.com/users')
+      .then((res) => res.json())
+      .then((json) => this.setState({ monsters: json }));
```
문제없이 잘 작성해주셨지만 이 부분에서 들어오는 데이터는 JSON 데이터가 아니기 때문에 헷갈릴 여지가 있는 이름은 피하시는게 좋습니다 :)

jamessoun93 commented on src/2-monsters/index.js:
```diff
@@ -23,21 +23,33 @@ import "./index.css";
     // 필터링 로직
+    const { monsters, userInput } = this.state;
+    const filteredMonster = monsters.filter((monster) =>
+      monster.name.toLowerCase().includes(userInput)
```
더욱더 정확한 비교를 위해 비교대상인 monster.name 과 userInput 을 둘다 소문자로 변환해주는 것이 좋습니다~

SangkuOh committed:
ADD: 모든기능 추가 완료
- 몬스터 API 호출 후 state 저장 완료
- CardList 컴포넌트 monsters props로 monster 배열 전달 완료
- Card 컴포넌트에 이미지주소, name, email 경로 수정 완료
- CardList에서 map 함수로 Card 컴포넌트 리턴완료
- Monster 컴포넌트 hanldeChange 메소드 자식에게 props 전달 후 setState
기능 구현 완료
- 변수에 필터링 로직 넣어서 값 비교후 조건에 맞는 카드만 보이게 구현완료

jamessoun93 commented on src/Assignments/Monsters/students/11th/sangkuOH/Components/Card/Card.js:
```diff
@@ -15,18 +15,18 @@ import "./Card.scss";
-          src="https://live.staticflickr.com/4561/38054606355_26429c884f_b.jpg"
+          src={`https://robohash.org/${this.props.id}?set=set2&size=180x180`}
           alt=""
         />
-        <h2>Name</h2>
-        <p>Email</p>
+        <h2>{this.props.name}</h2>
+        <p>{this.props.email}</p>
```
👍 👍 👍

jamessoun93 commented on src/Assignments/Monsters/students/11th/sangkuOH/Components/CardList/CardList.js:
```diff
@@ -10,8 +9,21 @@ import "./CardList.scss";
 
 class CardList extends Component {
+
+
```
자바스크립트 파일에서 두줄이나 비워주실 필요는 없습니다! 구분을 위해 적용하신 부분이라면, 한줄만 비워주셔도 됩니다~

SangkuOh replied:
넵🙈

jamessoun93 commented on src/Assignments/Monsters/students/11th/sangkuOH/Components/CardList/CardList.js:
```diff
@@ -10,8 +9,21 @@ import "./CardList.scss";
+        name={name}
+        email={email}
+      />)}
+      
```
인덴테이션과 괄호 문법 확인해주세요!

jamessoun93 commented on src/Assignments/Monsters/students/11th/sangkuOH/Monsters.js:
```diff
@@ -21,21 +21,48 @@ import "./Monsters.scss";
   // 데이터 로딩
-
+  componentDidMount(){
+  fetch("https://jsonplaceholder.typicode.com/users")
+    .then(res => res.json())
+    .then(res => this.setState({monsters: res}))
```
componentDidMount 의 body 안에 있는 내용이니 꼭 indent 해주세요!

jamessoun93 commented on src/Assignments/Monsters/students/11th/sangkuOH/Monsters.js:
```diff
@@ -21,21 +21,48 @@ import "./Monsters.scss";
+  handleChange = (e) => {
+    this.setState({
+      userInput: e.target.value
+    }, this.monsterLists)
```
setState 의 두번째 인자로 this.monsterList 를 넘겨주신 이유가 있으신가요?

SangkuOh replied:
해당 userInput을 가지고 바로 this.monsterLists 함수를 작동하여 filteredMonster를 filter를 하기 위함입니다.

jamessoun93 replied:
넵 아래 리뷰와 같이 메소드 하나에서 해당 로직을 처리할 수 있는데 고민해보시겠어요? 🤔

jamessoun93 commented on src/Assignments/Monsters/students/11th/sangkuOH/Monsters.js:
```diff
@@ -21,21 +21,48 @@ import "./Monsters.scss";
   // SearchBox 에 props로 넘겨줄 handleChange 메소드 정의
 
+  monsterLists = () => {
+      this.setState({filteredMonster : this.state.monsters.filter(({name}) => name.toLowerCase().includes(this.state.userInput))});
```
filteredMonsters 를 state 에 저장하실 계획이시라면, 아예 handleChange 메소드에서 한번에 e.target.value 로 필터링을 하시는 것은 어떠신가요? 😉

SangkuOh replied:
넵 해당 부분 수정반영중입니다. 👍

jamessoun93 reviewed (changes requested):
리뷰 확인해주세요~

SangkuOh committed:
Refactor: 수정요구사항 반영 완료
- CardList.js 인덴테이션 수정 완료
- Monsters.js fetch함수 indent 완료
- Monster.js filterdMonster 함수 handlerChange로 규합

jamessoun93 commented on src/Assignments/Monsters/students/11th/sangkuOH/Monsters.js:
```diff
@@ -29,34 +27,23 @@ class Monsters extends Component {
     this.setState({
-      userInput: e.target.value
-    }, this.monsterLists)
+      userInput: e.target.value,
+      filteredMonster : this.state.monsters.filter(({name}) => name.toLowerCase().includes(e.target.value))
+    })
```
오오오 👍 👍 👍

jamessoun93 reviewed (approved):
제가 말씀드린 코드 스타일적인 문제가 여전히 조금 남아있기는 하지만, 이런 부분들은 추후 리뷰를 통해 배우고 익히실 부분이라서 이만 merge 를 하도록 하겠습니다. 

LGTM 👍 
수고하셨습니다 😉
